### PR TITLE
fix: fixed enum parsing for json data

### LINKED ISSUE #744: [BUG]: Kiota fails to deserialize response when it contains array of enums
Kiota fails to de-serialize the response when it contains this array field of enum in OpenAPI:

```yaml
xyz:
  type: array
  items:
    type: string
    enum:
      - foo
      - bar
      - baz
```

It generates code like this:
```js
"xyz": n => { instance.xyz = n.getEnumValues(xyz); }
```

It calls [getEnumValues method](https://github.com/microsoft/kiota-typescript/blob/main/packages/serialization/json/src/jsonParseNode.ts#L94-L100) of @microsoft/kiota-serialization-json package

and it errors on

```ts
return rawValues.split(",").map((x) => type[toFirstCharacterUpper(x)] as T);
//              ^ Here, as the rawValues is already an array and array doesn't have split method
```

### PR DESCRIPTION
**Pull request by salmannotkhan**
refactor: updated method name `getEnumValues` to `getCollectionOfEnumValues`

Resolves #744

### CODE REVIEW

**Comment by salmannotkhan** [comment]
@microsoft-github-policy-service agree

File: packages/serialization/json/src/jsonParseNode.ts [hunk 1]
```diff
@@ -91,19 +91,19 @@ export class JsonParseNode implements ParseNode {
...
-    const rawValues = this.getStringValue();
-    if (!rawValues) {
-      return [];
+  public getCollectionOfEnumValues = <T>(type: any): T[] => {
+    if (Array.isArray(this._jsonNode)) {
+      const values = this._jsonNode.map(x => {
+        const node = new JsonParseNode(x)
+        return node.getEnumValue(type) as T
+      })
+      return values.filter(Boolean)
```
Comment (baywet) [inline packages/serialization/json/src/jsonParseNode.ts:100]:
```suggestion
      return this._jsonNode.map(x => {
        const node = new JsonParseNode(x)
        return node.getEnumValue(type) as T
      }).filter(Boolean);
```

File: packages/serialization/json/src/jsonParseNode.ts [hunk 2]
```diff
@@ -91,19 +91,19 @@ export class JsonParseNode implements ParseNode {
...
+    return []
   };
   public getEnumValue = <T>(type: any): T | undefined => {
-    const values = this.getEnumValues(type);
-    if (values.length > 0) {
-      return values[0] as T;
-    } else {
-      return undefined;
-    }
+    const rawValue = this.getStringValue()
```
Comment (baywet) [inline packages/serialization/json/src/jsonParseNode.ts:105]: this should test whether the string value is undefined or empty

File: packages/serialization/json/src/jsonParseNode.ts [hunk 3]
```diff
@@ -91,19 +91,19 @@ export class JsonParseNode implements ParseNode {
...
+    const value = type[toFirstCharacterUpper(rawValue)]
+    return value
```
Comment (baywet) [inline packages/serialization/json/src/jsonParseNode.ts:107]:
```suggestion
    return type[toFirstCharacterUpper(rawValue)];
```

**Review by baywet — CHANGES_REQUESTED** [review]
Thanks for the contribution here are a couple of suggestions.
Did you take the time to check whether the serialization writer was aligned in terms of logic as well?

**Comment by salmannotkhan** [comment]
I can create a PR for that if you want

**Comment by baywet** [comment]
I was referring to [this](https://github.com/microsoft/kiota-typescript/blob/78bfddff68f35306817456b0dc7d5ac922b6f5b8/packages/serialization/json/src/jsonSerializationWriter.ts#L139)
and [that](https://github.com/microsoft/kiota-serialization-json-dotnet/blob/295cfe1218b09d452e71443022d9c9ddcd7611c5/src/JsonSerializationWriter.cs#L289) (the later somehow missing from TypeScript???).
But yes, you can also go ahead with the generation PR.

**Comment by salmannotkhan** [comment]
That's what I was thinking about lol. But it works find even without that. I tested enum array before renaming the method. Do I have to change anything else in this repo?

**Comment by baywet** [comment]
Can you at least add an additional unit test that demonstrates symmetry please?

**Comment by salmannotkhan** [comment]
Did you mean [this](https://github.com/microsoft/kiota-typescript/pull/752/files#diff-1f2a5d21439bdf7808a752b2ec5d2f85776d137d51cb7986a32f068a4056a669)?

**Comment by salmannotkhan** [comment]
If you want me to add anything else, feel free to mention it

**Comment by baywet** [comment]
This is a deserialization test. I mean a test where you'd serialize and deserialize and compare the results.

**Comment by salmannotkhan** [comment]
Okay let me check

**Comment by salmannotkhan** [comment]
I think you mean [this](https://github.com/microsoft/kiota-typescript/blob/main/packages/serialization/json/test/common/jsonSerializationWriter.ts) but we don't have any method for `writeCollectionOfEnumValues` in `jsonSerializationWriter` maybe because it's just a normal string array.

**Comment by baywet** [comment]
yes, as far as I can remember they are treated as scalar values. Thanks for double checking

**Review by baywet — DISMISSED** [review]

**Pull request merged**